### PR TITLE
Smart Agent log receiver

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
(In lieu of https://github.com/signalfx/splunk-otel-collector/pull/149)

These changes allow the Smart Agent receiver to act as both a metrics and logs receiver, as well as removing the notion of an event client in config and Output functionality.  Instead they use standard log pipelines after I discovered that the main user of events in the agent, the processlist monitor, requires a host dimension that must be added via processor in a log pipeline unless hardcoded in agent config (https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/2631 is necessary in SFx exporter for this to take effect).

Also includes the basic SFx exporter dimension client defaulting logic, which is simpler without event clients.

### CODE REVIEW

File: internal/receiver/smartagentreceiver/README.md [hunk 1]
```diff
@@ -19,30 +19,58 @@ should be used.
...
+to use a Resource Detection to ensure that host identity and other useful information is made available as event
+dimensions.
 
 Example:
 
 ```yaml
-exporters:
-  signalfx:
+extensions:
+  smartagent:
```
Comment (tigrannajaryan) [inline internal/receiver/smartagentreceiver/README.md:39]: Is the extension required even if we don't specify any settings for it?

File: internal/receiver/smartagentreceiver/README.md [hunk 2]
```diff
@@ -19,30 +19,58 @@ should be used.
...
+
+service:
+  extensions: [smartagent]
+  pipelines:
+    metrics:
+      receivers: [smartagent/postgresql, smartagent/kafka]
+      processors: [resourcedetection]
+      exporters: [signalfx]
+    logs:
+      receivers: [smartagent/processlist]
```
Comment (tigrannajaryan) [inline internal/receiver/smartagentreceiver/README.md:71]: So, just to clarify. The logs pipeline is required for events to work, correct?
Do we want to document which monitors require it?

Comment (rmfitzpatrick) [reply internal/receiver/smartagentreceiver/README.md:71]: Yes, it's required and I will provide a list of dimension update and event using monitors w/ my next full documentation PR.

File: internal/receiver/smartagentreceiver/README.md (hunk 1 above)
Comment (rmfitzpatrick) [reply internal/receiver/smartagentreceiver/README.md:39]: Not necessary unless it changes settings.  Will remove for now.

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**